src/main/java/org/testng/TestNG.java

```diff
@@ -2057,7 +2065,7 @@ public void setGroupByInstances(boolean b) {
   //
 
   private URLClassLoader m_serviceLoaderClassLoader;
-  private List<ITestNGListener> m_serviceLoaderListeners = Lists.newArrayList();
+  private Map<Class<? extends ITestNGListener>, ITestNGListener> m_serviceLoaderListeners = Maps.newHashMap();
```

Comment:
Everywhere, when possible. I think it is easier to read/understand when the file is big. It assures you the variable won't be changed somewhere.